kolibri/core/assets/src/vue/content-renderer/index.vue

```diff
@@ -190,13 +190,20 @@
           // Add the specified options for the Vue component that we received from the plugin
           // into the options object.
           Object.assign(options, this.currentViewClass);
-          // Instantiate the Vue instance directly using the Kolibri Vue constructor.
-          this.contentView = new this.Kolibri.lib.vue(options); // eslint-disable-line new-cap
 
-          this.contentView.$on('startTracking', this.wrappedStartTracking);
-          this.contentView.$on('stopTracking', this.wrappedStopTracking);
-          this.contentView.$on('progressUpdate', this.wrappedUpdateProgress);
-          this.initSession(this.Kolibri, this.channelId, this.contentId, this.kind);
+          Promise.all(
+            // guarantee summarylog, sessionlog, and existing masterylog are synced and in store.
+            this.initSession(this.Kolibri, this.channelId, this.contentId, this.kind)
+          ).then(() => {
```

Comment:
As mentioned above, if `Promise.all` can be moved inside the `initContentSession` action, and then this becomes

```javascript
          // guarantee summarylog, sessionlog, and existing masterylog are synced and in store.
          this.initSession(this.Kolibri, this.channelId, this.contentId, this.kind).then(() => {
```